backbone.nativeajax.js

```diff
@@ -16,18 +16,52 @@
   //   var req = Backbone.ajax({url: 'url', type: 'PATCH', data: 'data'});
   //   req.then(..., ...) // if Promise is set
   var ajax = (function() {
-    var xmlRe = /^(?:application|text)\/xml/;
-    var jsonRe = /^application\/json/;
-
-    var getData = function(accepts, xhr) {
-      if (accepts == null) accepts = xhr.getResponseHeader('content-type');
-      if (xmlRe.test(accepts)) {
-        return xhr.responseXML;
-      } else if (jsonRe.test(accepts) && xhr.responseText !== '') {
-        return JSON.parse(xhr.responseText);
-      } else {
-        return xhr.responseText;
+    var xmlRe = /^(?:application|text)\/xml$/;
+    var jsonRe = /^application\/json$/;
+
+    var parseMimeType = function(mimeType) {
+        if (xmlRe.test(mimeType)) {
+          return 'xml';
+        } else if (jsonRe.test(mimeType)) {
+          return 'json';
+        }
+    }
+
+    var getContentType = function(options, xhr) {
+      var contentType = options.dataType;
+      if (!contentType) {
+        var contentTypeHeader = xhr.getResponseHeader('Content-type');
+        contentType = parseMimeType(contentTypeHeader);
+      }
+      if (!contentType) {
+        var acceptHeader = options.headers && options.headers.Accept;
+        var accepts = acceptHeader.replace(/;[^,]*/g, '').split(/\s*,\s*/);
+        for (var i=0; i<accepts.length; i++) {
+          contentType = parseMimeType(accepts[i]);
+          if (contentType) {
+            break;
+          }
+        }
+      }
+      return contentType;
+    }
+
+    var getData = function(options, xhr) {
+      var contentType = getContentType(options, xhr);
+      var data;
+      switch (contentType) {
+        case 'xml':
+          data = xhr.responseXML;
+          break;
+        case 'json':
+          if (xhr.responseText !== '') {
```

Comment:
I copied this check from https://github.com/akre54/Backbone.NativeAjax/blob/master/backbone.nativeajax.js#L26. I don't know what it's for.